tests/backend_test.py

```diff
@@ -52,19 +62,148 @@ def test_quantinuum_sim_h11e(azure_backend: AzureBackend) -> None:
+        h = a_b.process_circuit(c1, n_shots=1000)
+        r = a_b.get_result(h, timeout=120)
+        counts = r.get_counts()
+        assert sum(counts.values()) == 1000
+    else:
+        raise ValueError(
+            "quantinuum.sim.h1-1sc unavailable or queue time >= 60s: not submitting"
+        )
+
+
+@pytest.mark.skipif(skip_remote_tests, reason=REASON)
+@pytest.mark.parametrize("azure_backend", ["quantinuum.sim.h1-1sc"], indirect=True)
+def test_quantinuum_sim_h11e_cond_2(azure_backend: AzureBackend) -> None:
+    c = Circuit(1, name="test_classical")
+    a = c.add_c_register("a", 32)
+    b = c.add_c_register("b", 32)
+    d = c.add_c_register("d", 32)
+
+    c.Measure(Qubit(0), a[0])
+
+    c.add_c_setreg(23, b)
+
+    c.add_classicalexpbox_register(a + b, d)  # type: ignore
+    c.add_classicalexpbox_register(a - b, d)  # type: ignore
+    c.add_classicalexpbox_register(a << 1, a)  # type: ignore
+    c.add_classicalexpbox_register(a >> 1, b)  # type: ignore
+
+    c.X(0, condition=reg_eq(a ^ b, 1))
+    c.X(0, condition=(a[0] ^ b[0]))
+    c.X(0, condition=reg_eq(a & b, 1))
+    c.X(0, condition=reg_eq(a | b, 1))
+
+    c.X(0, condition=a[0])
+    c.Measure(Qubit(0), b[4])
+
+    c.X(0, condition=reg_neq(a, 1))
+    c.X(0, condition=if_not_bit(a[0]))
+    c.X(0, condition=reg_gt(a, 1))
+    c.X(0, condition=reg_lt(a, 1))
+    c.X(0, condition=reg_geq(a, 1))
+    c.X(0, condition=reg_leq(a, 1))
+    c.Measure(Qubit(0), b[4])
+    a_b = azure_backend
+    c1 = a_b.get_compiled_circuit(c)
+    if a_b.is_available() and a_b.average_queue_time_s() < 60:
+        h = a_b.process_circuit(c1, n_shots=1000)
+        r = a_b.get_result(h, timeout=120)
+        counts = r.get_counts()
+        assert sum(counts.values()) == 1000
+    else:
+        raise ValueError(
+            "quantinuum.sim.h1-1sc unavailable or queue time >= 60s: not submitting"
+        )
 
 
 @pytest.mark.skipif(skip_remote_tests, reason=REASON)
 @pytest.mark.parametrize("azure_backend", ["quantinuum.sim.h1-1e"], indirect=True)
 def test_quantinuum_option_params(azure_backend: AzureBackend) -> None:
-    c = Circuit(2).H(0).CX(0, 1).measure_all()
-    b = azure_backend
-    c1 = b.get_compiled_circuit(c)
-    if b.is_available() and b.average_queue_time_s() < 600:
-        h = b.process_circuit(c1, n_shots=1000, option_params={"error_model": False})  # type: ignore
-        r = b.get_result(h, timeout=1200)
+    c = Circuit(2, 2).H(0).CX(0, 1).measure_all()
+    a_b = azure_backend
+    c1 = a_b.get_compiled_circuit(c)
+    if a_b.is_available() and a_b.average_queue_time_s() < 600:
+        h = a_b.process_circuit(c1, n_shots=1000, option_params={"error_model": False})  # type: ignore
+        r = a_b.get_result(h, timeout=1200)
         counts = r.get_counts()
-        assert all(x0 == x1 for x0, x1 in counts)
+        assert all(x[0] == x[1] for x in counts)
+        assert any(x[0] == 1 for x in counts)
```

Comment:
This is likely but not guaranteed.

Reply:
I thought this is a valid assumption for 1000 shots. If you want me to remove this, I will remove it.